xalia/Uia/MsaaElement.cs

```diff
@@ -49,6 +60,61 @@ public MsaaElement(MsaaElementWrapper wrapper, UiaConnection root) : base(root)
         private bool _roleKnown;
         private bool _fetchingRole;
 
+        private int _state;
+        private bool _stateKnown;
+        private bool _fetchingState;
+        private CancellationTokenSource _fetchingStateCancel;
+
+        internal static string[] msaa_state_names =
+        {
+            "unavailable",
+            "selected",
+            "focused",
+            "pressed",
+            "checked",
+            "mixed",
+            "readonly",
+            "hottracked",
+            "default",
+            "expanded",
+            "collapsed",
+            "busy",
+            "floating",
+            "marqueed",
+            "animated",
+            "invisible",
+            "offscreen",
+            "sizeable",
+            "moveable",
+            "selfvoicing",
+            "focusable",
+            "selectable",
+            "linked",
+            "traversed",
+            "multiselectable",
+            "extselectable",
+            "alert_low",
+            "alert_medium",
+            "alert_high",
+            "protected",
+        };
+
+        internal static Dictionary<string, int> msaa_name_to_state;
+
+        protected override void SetAlive(bool value)
+        {
+            if (!value)
+            {
+                _fetchingState = false;

```

Comment:
State fetching is able to be cancelled, but role fetching is not. I assume this is deliberate because role is likely only fetched once while state is continuously queried, but worth pointing out just to be sure.

Reply:
This is deliberate because there's no ROLECHANGE event. Resolving.